vulcanai2/models/basenetwork.py

```diff
@@ -144,8 +144,7 @@ def __init__(self, name, config, in_dim=None, save_path=None,
         # Compute self.out_dim of the network
         self.out_dim = self._get_out_dim()
 
-        self._device = torch.device(device if torch.cuda.is_available() else "cpu")
-        self.to(device=self._device)
```

Comment:
why remove this? when you set device don't you have to cast the network to the device? It's causing issues now

Reply:
because we had duplicated the setter function in the init. I just simplified it by calling the device setter which runs those lines of code. Would you be able to provide the exact error/issue that you're running into?

Reply:
It used to work this way:
when we set `multi_input_conv_3D.device = "cpu"`
it will set to via  `multi_input_conv_3D.input_networks[#].device = "cpu"`, remember I showed you the other day?
Now, it doesn't seem to be working. I have to loop into individual input_networks in this case. Just came across this when testing with a loaded model.

Reply:
you can try to add in the device init code back but honestly its the same code as the device.setter. however, I don't think we should set all input networks to the same device as the current network since users may have already specified them to a specific cpu and don't want it to be moved again. we can add an option to the setter to also change the input networks devices

Reply:
I think `torch.distributed` would do a better job if we are considering multiple device scenario.
Also, if you'd like to have the input_networks in the default device unless specified, how do you plan to assign the data's device based on the network's device? Only option I can think of is looping through each input_network and check for the device and assign that device to the incoming data, whichcan done within forward function but I'm not sure if that is recommended.

Reply:
torch.distributed is for training across various machines. we are still on the same machine with different gpus. I just delayed the data device casting until the forward function which seems to work pretty well especially if users change network gpu devices in real-time

Reply:
I mean dataparallel. It looks like the suggested way is to assign the device to data in forward function: https://pytorch.org/tutorials/beginner/former_torchies/parallelism_tutorial.html